test/unit-test/cellular_common_api_utest.c

```diff
@@ -0,0 +1,921 @@
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    cellularStatus = Cellular_CommonRegisterUrcGenericCallback( ( CellularHandle_t ) &context, NULL, NULL );
+
+    TEST_ASSERT_EQUAL( CELLULAR_SUCCESS, cellularStatus );
+}
+
+/**
+ * @brief Test that null handler case for Cellular_CommonRegisterModemEventCallback.
+ */
+void test_Cellular_CommonRegisterModemEventCallback_Null_Handler( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_INVALID_HANDLE );
+
+    cellularStatus = Cellular_CommonRegisterModemEventCallback( NULL, NULL, NULL );
+
+    TEST_ASSERT_EQUAL( CELLULAR_INVALID_HANDLE, cellularStatus );
+}
+
+/**
+ * @brief Test that happy path case for Cellular_CommonRegisterModemEventCallback.
+ */
+void test_Cellular_CommonRegisterModemEventCallback_Happy_Path( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    cellularStatus = Cellular_CommonRegisterModemEventCallback( ( CellularHandle_t ) &context, NULL, NULL );
+
+    TEST_ASSERT_EQUAL( CELLULAR_SUCCESS, cellularStatus );
+}
+
+/**
+ * @brief Test that null handler case for Cellular_CommonATCommandRaw.
+ */
+void test_Cellular_CommonATCommandRaw_Null_Handler( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_INVALID_HANDLE );
+
+    cellularStatus = Cellular_CommonATCommandRaw( NULL, NULL, NULL, 0, NULL, NULL, 0 );
+
+    TEST_ASSERT_EQUAL( CELLULAR_INVALID_HANDLE, cellularStatus );
+}
+
+/**
+ * @brief Test that null at command payload case for Cellular_CommonATCommandRaw.
+ */
+void test_Cellular_CommonATCommandRaw_Null_AtCmdPayload( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    cellularStatus = Cellular_CommonATCommandRaw( &context, NULL, NULL, 0, NULL, NULL, 0 );
+
+    TEST_ASSERT_EQUAL( CELLULAR_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test that happy path case for Cellular_CommonATCommandRaw.
+ */
+void test_Cellular_CommonATCommandRaw_Happy_Path( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+    char pPrefix[] = "AtTSest";
+    char pData[] = "Test Data";
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    _Cellular_AtcmdRequestWithCallback_IgnoreAndReturn( CELLULAR_PKT_STATUS_OK );
+    _Cellular_TranslatePktStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    cellularStatus = Cellular_CommonATCommandRaw( &context, pPrefix, pData, 0, NULL, NULL, 0 );
+
+    TEST_ASSERT_EQUAL( CELLULAR_SUCCESS, cellularStatus );

```

Comment:
Should we assert that `_Cellular_AtcmdRequestWithCallback` was supplied the correct parameter?

Reply:
Sure, I can add one case for this purpose.